src/libsyntax_pos/lib.rs

```diff
@@ -83,16 +83,49 @@ pub struct MultiSpan {
 }
 
 impl Span {
+    #[inline]
+    pub fn new(lo: BytePos, hi: BytePos, ctxt: SyntaxContext) -> Self {
+        Span { lo, hi, ctxt }
+    }
+
+    #[inline]
+    pub fn lo(self) -> BytePos {
+        self.lo
+    }
+    #[inline]
+    #[must_use]
+    pub fn set_lo(self, lo: BytePos) -> Span {
```

Comment:
Should the fields be named `with_x`? In my mind that conjures the fact that the method returns a copy of the same type with the field having the new value.